benchmarks/helper.py

```diff
@@ -0,0 +1,222 @@
+
+where
+    e2e - generates the commands, runs the benchmarks, parse the results (other options - commands/parse)
+    nightly - Looks at the nightly field in the training/inferece field of TABLE to choose most relevant compilers.
+    models - either training or inference
+    suites - benchmark suite
+
+In case, you just need to get the commands, you can replace e2e with commands
+-> python benchmarks/helper.py --commands --modes=training --devices=cuda --suites=torchbench --dtypes=float32 --nightly
+
+The commands are written in file run.sh.
+
+In case, you just want to parse the logs once you have run the commands manually, you can use this command
+-> python benchmarks/helper.py --parse --modes=training --devices=cuda --suites=torchbench --dtypes=float32 --nightly
+
+"""
+
+import argparse
+import itertools
+import os
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib import rcParams
+from tabulate import tabulate
+
+rcParams.update({"figure.autolayout": True})
+plt.rc("axes", axisbelow=True)
+
+DEFAULT_OUTPUT_DIR = "benchmark_logs"
+
+
+TABLE = {
+    "suites": ["torchbench", "huggingface"],
+    "dtypes": ["float32", "float16", "amp"],
+    "devices": ["cuda", "cpu"],
+    "modes": ["inference", "training"],
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "training": {
+        "ts_nnc": "--training --speedup-ts --use-eval-mode --isolate",
+        "ts_nvfuser": "--training --nvfuser --speedup-ts --use-eval-mode --isolate",
+        "aot_eager": "--training --accuracy-aot-nop --generate-aot-autograd-stats --use-eval-mode --isolate",
+        "aot_nnc": "--training --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "aot_nvfuser": "--training --nvfuser --accuracy-aot-ts-mincut --use-eval-mode --isolate",
+        "nightly": ["ts_nvfuser", "aot_nvfuser"],
+    },
+    # Dict of name to base_command. nightly is a special field to tell which compilers are relevant.
+    "inference": {
+        "ts_nnc": "-dcuda --isolate --speedup-ts",
+        "ts_nvfuser": "-dcuda --isolate -n100 --speedup-ts --nvfuser",
+        "trt": "-dcuda --isolate -n100 --speedup-trt",
+        "eager_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs",
+        "nnc_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts --nvfuser",
+        "nvfuser_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --backend=cudagraphs_ts",
+        "inductor_cudagraphs": "-dcuda --inductor-settings --float32 -n50 --inductor",
+        "nightly": ["nvfuser_cudagraphs", "inductor_cudagraphs"],
+    },
+}
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--filter", "-k", action="append", help="filter benchmarks with regexp"
+    )
+    parser.add_argument("--devices", action="append", help="cpu or cuda")
+    parser.add_argument("--modes", action="append", help="inference or training")
```

Comment:
maybe cut out 'modes' arg,

add --training and --inference args which are both append actions,  

have the default for training = ["ts_nvfuser", "aot_nvfuser"]
get rid of the 'nightly' flag altogether since now it is just the 'default' for --training

and it could be an error if you specify both --training and --inference or it could work to specify both, depending how hard that is to implement?

Reply:
Perfect. This is much better. Thanks!